### PR TITLE
Add extra config for white list codec.

### PR DESCRIPTION
**Pull request by zhangyihui1**

### CODE REVIEW

File: sdk/android/api/org/webrtc/HardwareVideoDecoderFactory.java [hunk 1]
```diff
@@ -132,21 +133,25 @@ private boolean isHardwareSupported(MediaCodecInfo info, VideoCodecType type) {
         return name.startsWith(QCOM_PREFIX) || name.startsWith(INTEL_PREFIX)
             || name.startsWith(EXYNOS_PREFIX) || name.startsWith(NVIDIA_PREFIX)
             // Hisi seems to support VP8.
-            || name.startsWith(HISI_PREFIX);
+            || name.startsWith(HISI_PREFIX)
+            || vcp.isExtraHardwareSupported(name, "video/x-vnd.on2.vp8", vcp.parseWithTag(vcp.loadWithDom(extraMediaCodecFile), "Decodeds"));
```
Comment (taste1981) [inline sdk/android/api/org/webrtc/HardwareVideoDecoderFactory.java]: "Decoders" not "Decodeds".  Please also check your xml setting.

File: sdk/android/api/org/webrtc/HardwareVideoEncoderFactory.java [hunk 2]
```diff
@@ -208,7 +212,8 @@ private boolean isHardwareSupportedInCurrentSdkH264(MediaCodecInfo info) {
         || (name.startsWith(EXYNOS_PREFIX)
                && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)
         // Hisi H264 encoder seems to be supported. Needs more testing.
-        || (name.startsWith(HISI_PREFIX) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT);
+        || (name.startsWith(HISI_PREFIX) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT)
+        || (vcp.isExtraHardwareSupported(name, "video/avc", vcp.parseWithTag(vcp.loadWithDom(extraMediaCodecFile), "Encoders")) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT);
   }
 
   private boolean isHardwareSupportedInCurrentSdkH265(MediaCodecInfo info) {
```
Comment (taste1981) [inline sdk/android/api/org/webrtc/HardwareVideoEncoderFactory.java:219]: you should be doing the same for hevc.

File: sdk/android/api/org/webrtc/VideoCapabilityParser.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright 2017 The WebRTC project authors. All Rights Reserved.
```
Comment (taste1981) [inline sdk/android/api/org/webrtc/VideoCapabilityParser.java]: 2019

File: sdk/android/api/org/webrtc/HardwareVideoDecoderFactory.java (hunk 1 above)
Comment (zhangyihui1) [reply sdk/android/api/org/webrtc/HardwareVideoDecoderFactory.java]: done

File: sdk/android/api/org/webrtc/VideoCapabilityParser.java (hunk 3 above)
Comment (zhangyihui1) [reply sdk/android/api/org/webrtc/VideoCapabilityParser.java]: done

File: sdk/android/api/org/webrtc/HardwareVideoEncoderFactory.java (hunk 2 above)
Comment (zhangyihui1) [reply sdk/android/api/org/webrtc/HardwareVideoEncoderFactory.java:219]: done

**Review by taste1981 — APPROVED** [review]
Make sure you add document in android sdk, providing a sample xml file for reference and document where to put that file.

**Pull request merged**